packages/lib/server/getLuckyUser.ts

```diff
@@ -364,19 +364,350 @@ export async function getLuckyUser<
+    allRRHostsBookingsOfThisMonth,
+    allRRHostsCreatedThisMonth,
+    organizersWithLastCreated,
+  ] = await Promise.all([
+    getCurrentMonthsBookings({
+      eventTypeId: eventType.id,
+      users: availableUsers.map((user) => {
+        return { id: user.id, email: user.email };
+      }),
+      virtualQueuesData,
+    }),
+
+    getCurrentMonthsBookings({
+      eventTypeId: eventType.id,
+      users: notAvailableHosts,
+      virtualQueuesData,
+    }),
+
+    getCurrentMonthsBookings({
+      eventTypeId: eventType.id,
+      users: allRRHosts.map((host) => {
+        return { id: host.user.id, email: host.user.email };
+      }),
+      virtualQueuesData,
+    }),
+
+    prisma.host.findMany({
+      where: {
+        userId: {
+          in: allRRHosts.map((host) => host.user.id),
+        },
+        eventTypeId: eventType.id,
+        isFixed: false,
+        createdAt: {
+          gte: startOfMonth,
+        },
+      },
+    }),
+
+    prisma.user.findMany({
+      where: {
+        id: {
+          in: availableUsers.map((user) => user.id),
+        },
+      },
+      select: {
+        id: true,
+        bookings: {
+          select: {
+            createdAt: true,
+          },
+          where: {
+            eventTypeId: eventType.id,
+            status: BookingStatus.ACCEPTED,
+            attendees: {
+              some: {
+                noShow: false,
+              },
+            },
+            // not:true won't match null, thus we need to do an OR with null case separately(for bookings that might have null value for `noShowHost` as earlier it didn't have default false)
+            // https://github.com/calcom/cal.com/pull/15323#discussion_r1687728207
+            OR: [
+              {
+                noShowHost: false,
+              },
+              {
+                noShowHost: null,
+              },
+            ],
+          },
+          orderBy: {
+            createdAt: "desc",
+          },
+          take: 1,
+        },
+      },
+    }),
+  ]);
+
+  const endTime = performance.now();
+  log.info(`fetchAllDataNeededForCalculations took ${endTime - startTime}ms`);
+
+  console.log({
+    currentMonthBookingsOfAvailableUsers: currentMonthBookingsOfAvailableUsers.length,
+    bookingsOfNotAvailableUsersOfThisMonth: bookingsOfNotAvailableUsersOfThisMonth.length,
+    allRRHostsBookingsOfThisMonth: allRRHostsBookingsOfThisMonth.length,
+    allRRHostsCreatedThisMonth: allRRHostsCreatedThisMonth.length,
+  });
+
+  return {
+    currentMonthBookingsOfAvailableUsers,
+    bookingsOfNotAvailableUsersOfThisMonth,
+    allRRHostsBookingsOfThisMonth,
+    allRRHostsCreatedThisMonth,
+    organizersWithLastCreated,
     attributeWeights,
-    virtualQueuesData
+    virtualQueuesData,
+  };
+}
+
+type AvailableUserBase = PartialUser & {
+  priority?: number | null;
+  weight?: number | null;
+};
+
+export async function getOrderedListOfLuckyUsers<AvailableUser extends AvailableUserBase>(
```

Comment:
we also need to pass  the `routingFormResponse` here so we can get `attributeWeights` and `virtualQueuesData`